### ISSUE TITLE
Connection to tor nodes

### ISSUE THREAD

**Issue by FlorianHubl** [lightningdevkit/ldk-node#178]
Is there a way to connect to lightning nodes which run on tor? I run tor in a iOS app. How can I set the proxy in ldk-node?

**Comment by tnull** [comment]
> Would love to see this, and the ability to query the Esplora nodes over Tor too

Noted. I now opened a separate issue as first we'd need to add a separate networking crate to `rust-lightning`: https://github.com/lightningdevkit/rust-lightning/issues/4299

**Comment by tnull** [comment]
> I think with [lightningdevkit/rust-lightning#4305](https://github.com/lightningdevkit/rust-lightning/pull/4305) this could be unblocked?

Yes, indeed, it should be.

> I have a node based on 0.7.0 that successfully connected over Tor, with this patch:
> 
> [jharveyb@ab23a78](https://github.com/jharveyb/ldk-node/commit/ab23a78597a3da87d84cee8a073511e81791a516)

Awesome, already looks pretty good! Do you want to open a PR for this?

> One awkward bit is that the peer is stored internally with the tor proxy socketaddress, not the original onionv3. Will keep testing.

Ah, alright. Happy to brainstorm/discuss options on the PR.
